gateway/src/apicast/policy/3scale_batcher/backend_downtime_cache.lua

```diff
@@ -0,0 +1,31 @@
+--- Backend downtime cache
+-- Cache to be used when the caching policy is in the policy chain.
+-- The caching policy exposes a 'cache_handler' method. This class relies
+-- on that method to update the cache instead of providing its own one.
+
+local setmetatable = setmetatable
+local keys_helper = require 'apicast.policy.3scale_batcher.keys_helper'
+
+local _M = {}
+
+local mt = { __index = _M }
+
+function _M.new(shared_dict)
+  local self = setmetatable({}, mt)
+  self.shared_dict = shared_dict
+  return self
+end
+
+function _M:get(transaction)
+  local key = keys_helper.key_for_cached_auth(transaction)
+  return self.shared_dict:get(key)
+end
+
+-- Note: "update_func" should be one of the handlers exposed by the caching
+-- policy.
+function _M:update(transaction, backend_status, update_func)
```

Comment:
I'll move `update_func` to the initializer so it becomes an attr of the class.

Reply:
What about not passing `update_func` at all?

It is passed here only to be called. This function just gets keys for a transaction. 

Actually this class feels a bit weird as it has several collaborators, but they are grouped only by one property: they need keys for the transaction. One reads from shdict and other is a function to be executed. Seems to me they don't really belong to be together.

Reply:
You're right. I was having second thoughts about this class when I opened this PR, and that's why I added the comment. The objective was to isolate the management of the backend downtime cache, but we need to use a handler stored in the context and that makes it a bit weird.

In the end, I left the cache handling responsibility in the main class of the policy. I think that's better for now.